src/sisl/io/siesta/stdout.py

```diff
@@ -1164,7 +1164,168 @@ def _parse_charge(line):
+                # for each species are enclosed by dashes (-----)
+
+                # Step to header
+                _, line = self.step_to("Atom", allow_reread=False)
+                if header is None:
+                    header = (
+                        line.replace("Charge", "e")
+                        .replace("Spin", "S")
+                        .replace(
+                            "Svec", "Sx Sy Sz"
+                        )  # Split Svec into Cartesian components
+                        .split()
+                    )[2:]
+
+                # Skip over the starting ---- line
+                self.readline()
+
+                # Read until closing ---- line
+                line = ""
+                while "----" not in line:
+                    line = self.readline()
+                    if "Total" in line:
+                        ia, charge_vals = _parse_charge_total_nc(line)
+                        atom_idx.append(ia)
+                        atom_charges.append(charge_vals)
+
+            def _parse_spin_pol():  # unpolarized and colinear spin
+                nonlocal atom_idx, atom_charges
+
+                # The mulliken charges are organized per spin
+                # polarization (UP/DOWN). The end of a spin
+                # block is marked by a Qtot
+
+                # Read until we encounter "mulliken: Qtot"
+                def try_parse_int(s):
+                    try:
+                        int(s)
+                    except ValueError:
+                        return False
+                    else:
+                        return True
+
+                line = ""
+                while "mulliken: Qtot" not in line:
+                    line = self.readline()
+                    words = line.split()
+                    if len(words) > 0 and try_parse_int(words[0]):
+                        # This should be a line containing the total charge for an atom
+                        ia, charge = _parse_charge_total(line)
+                        atom_idx.append(ia)
+                        atom_charges.append(charge)
+
+            # Determine with which spin type we are dealing
+            IDX_NON_POL = 0
+            IDX_POL = 1
+            IDX_NC = 2
+            search_keys_spin = [
+                "Qatom",  # if we find Qatom and no "mulliken: Spin UP" we are dealing with spin unpolarized
+                "mulliken: Spin UP",
+                "Svec",
+            ]
+            loc = self.fh.tell()
+            ret = self.step_to(
+                search_keys_spin, case=True, ret_index=True, allow_reread=False
+            )
+            self.fh.seek(loc)
+            if ret[2] == IDX_NON_POL:
+                # No spin components so just parse charge
+                atom_charges = []
+                atom_idx = []
+                header = ["e"]
+                _parse_spin_pol()
+            elif ret[2] == IDX_POL:
```

Comment:
I don't know if it makes sense to use `self.info.spin == Spin.POLARIZED` instead? The `self.info.spin` should be populated well before any mulliken charges is created.

Reply:
Ah, I didn't know that. If that works that would certainly be preferable. I'll give it a try.

Reply:
Are you sure `self.info.spin` gets populated? I tried this but I get,

```
AttributeError: stdoutSileSiesta.info.spin does not exist, did you mistype?
```

Reply:
I've added an additional `spin` property to the `InfoAttr` dict at the start of `stdOutSileSiesta` and now it works (see commit https://github.com/zerothi/sisl/pull/691/commits/fece8347095586e3b00fa08ba65beaa65b6ba310). Was this sort of what you had in mind?

Reply:
ha, sorry, I had added that commit just after you branched off, see here https://github.com/zerothi/sisl/blob/13a67c772bb3e612542c81d7549038bc410e9ec3/src/sisl/io/siesta/stdout.py#L41

So basically we did the same thing, yes, it was exactly how I imagined it. If you could revert your commit, then rebase, or ask me to rebase :)

Reply:
I reverted my last commit to remove any changes I made to add the `self.info.spin` property and then rebased on the latest version of `main`. Everything should be okay again now :)

I still have to add the tests to finalize this PR but I'll do my very best to allocate some time to do this next week.

Reply:
Sounds really great!

Reply:
In case you are not aware, files should go in the https://github.com/zerothi/sisl-files repository, and then the submodule `files` should be bumped, if you need help, let me know. :)

Reply:
I didn't get around to this unfortunately and I'll be away next week so I won't be able to finalize this until the first week of April. I hope this is okay.